File: 242-valid-anagram/valid-anagram.py
from collections import defaultdict
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        hashMap = defaultdict(int)

        if len(s) != len(t):
            return False

        for ch in s:
            hashMap[ch] = hashMap.get(ch,0) +1
        print(hashMap)

        for ch in t:
            if ch not in hashMap:
                return False
            hashMap[ch] = hashMap.get(ch,0) -1

            if hashMap[ch]  <= 0:
                del hashMap[ch]
        
        return True 

        
        

File: 242-valid-anagram/valid-anagram.cpp
#include <unordered_map>
class Solution {
public:
    bool isAnagram(string s, string t) {
        std::unordered_map<char, int> map;

        for(auto x: s){
           map[x]++;
        }

        for(auto y:t){
            map[y]--;
        }

        for(auto x:map){
            if(x.second != 0){
                return false;
            }
        }

        return true;

        
    }
};